### PR TITLE
Added _UICache.h

### PR DESCRIPTION
**Pull request by dlevi309**
What does this implement/fix? Explain your changes.
---------------------------------------------------
… Interface for clearing UICache programmatically 

Does this close any currently open issues?
------------------------------------------
… No

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** … 13.4, 14.4

**Platform:** … iOS

**Target Platform:** … iOS

**Toolchain Version:** … LLVM 10.0.0 on sbingner's repo

**SDK Version:** … 13.4

### CODE REVIEW

File: UIKit/_UICache.h
Comment (leptos-null) [inline UIKit/_UICache.h:5]: Is this line necessary?

**Review by leptos-null — APPROVED** [review]
Looks good to me.
Uses `NS_CLASS_AVAILABLE_IOS` and `NS_AVAILABLE_IOS` as opposed to the more modern `API_AVAILABLE`, but the meaning should be clear

File: UIKit/_UICache.h
Comment (PoomSmart) [inline UIKit/_UICache.h:11]: Do we know what kind of data go into this array?

**Comment by dlevi309** [comment]
I just realized this class is actually apart of UIFoundation, maybe hold off on the merge until I (or anyone else) play around with it

File: UIKit/_UICache.h
Comment (dlevi309) [reply UIKit/_UICache.h:11]: I’m sorta stumped on it, I’d only tested the clearsCacheOnLowMemoryWarnings and clearsCacheOnApplicationBackground (which is always set to YES by default) I assume an example of this that is actively used is related to network cache (as that is always purged upon jumping to another app or the springboard)
Maybe file paths?

File: UIKit/_UICache.h
Comment (dlevi309) [reply UIKit/_UICache.h]: I suppose it is not, thanks, just edited it accordingly

File: UIKit/_UICache.h
Comment (chrisharper22) [reply UIKit/_UICache.h]: There are several class declarations such as `_UIFontCacheKey` and `_UIImageCacheKey` spread throughout frameworks (notably UIFoundation and UIKit). Maybe it refers to those?

Comment (dlevi309) [reply UIKit/_UICache.h]: @chrisharper22 I was actually looking at that as a possibility about 10 min ago, I don’t think so, only because the values that return calling them with FLEX seem related to images that are cached with a .car file (so when it refers to “cached” images, I believe it means images within the .car file and not actual device cache. Or if not, .cpbitmap also seems like it falls under the same logic, except those are built to be actual cached files of images (that applications don’t control, see `/var/mobile/Library/Caches/MappedImageCache` on a device for more info on what type of images are cached (wallpaper, clock hands, Battery Center stuff, etc.)

Comment (dlevi309) [reply UIKit/_UICache.h]: @chrisharper22 something else that supports my theory is that all the image types described (Keyboard images, Status Bar items, etc.) are the types of images found when extracting the assets from  `/System/Library/PrivateFrameworks/UIKitCore.framework/Artwork.bundle/Assets.car`

**Comment by dlevi309** [comment]
This was tested on with the App Store and FLEXing’s Network History tab

File: UIKit/_UICache.h
Comment (leptos-null) [reply UIKit/_UICache.h]: (@dlevi309 I'm pretty sure PoomSmart was asking about the data type, meaning is the type `NSArray<NSString *> *`, etc.)

File: UIKit/_UICache.h
Comment (leptos-null) [inline UIKit/_UICache.h:3]: I think you removed the class declaration here

Comment (dlevi309) [reply UIKit/_UICache.h:3]: 🤦🏻‍♂️

**Comment by PoomSmart** [comment]
> I just realized this class is actually apart of UIFoundation, maybe hold off on the merge until I (or anyone else) play around with it

If this is indeed in `UIFoundation` then it should be moved accordingly.

**Comment by dlevi309** [comment]
> > I just realized this class is actually apart of UIFoundation, maybe hold off on the merge until I (or anyone else) play around with it
> 
> If this is indeed in `UIFoundation` then it should be moved accordingly.

How're we lookin now?

**Comment by leptos-null** [comment]
Do we _want_ to import `UIKit`? I don't see anything from UIKit being used in the header

**Comment by dlevi309** [comment]
> Do we _want_ to import `UIKit`? I don't see anything from UIKit being used in the header

No, you're right. I was still under the impression that this class was originally a part of UIKit before iOS 11, but I just checked around and it never was.

File: UIKit/_UICache.h
Comment (PoomSmart) [reply UIKit/_UICache.h]: Explored those kind of objects on FLEX and found that this array isn't necessary of NSString. Omitting type is fine by me.

File: UIFoundation/_UICache.h
Comment (PoomSmart) [inline UIFoundation/_UICache.h:12]:
```suggestion
@property (nonatomic, assign, readwrite) BOOL clearsCacheOnLowMemoryWarnings;
```

File: UIFoundation/_UICache.h
Comment (PoomSmart) [inline UIFoundation/_UICache.h:14]:
```suggestion
@property (nonatomic, assign, readwrite) BOOL clearsCacheOnApplicationBackground;
```

File: UIFoundation/_UICache.h
Comment (PoomSmart) [inline UIFoundation/_UICache.h:7]: If you want, there's also `-retainedObjectForKey:`

**Comment by kirb** [comment]
@PoomSmart and @dlevi309, do you think we can merge this? If not could one of you click Resolve on all the review threads that are dealt with, and when you have a moment, address the others? (Or let me know what I can do to fix it up for you.) Thanks!

**Review by PoomSmart — APPROVED** [review]

**Pull request merged**
